backend/src/main/java/com/redhat/cloud/notifications/db/repositories/NotificationRepository.java

```diff
@@ -54,6 +54,31 @@ public List<NotificationHistory> getNotificationHistory(String orgId, UUID endpo
                 .getResultList();
     }
 
+    /**
+     * Counts the number of notification history elements for the given Endpoint ID and OrgId.
+     * @param endpointId the endpoint's ID the notifications will be filtered with.
+     * @param orgId the event's OrgId the notifications will be filtered with.
+     * @return the number of notification history elements.
+     */
+    public long countNotificationHistoryElements(UUID endpointId, String orgId) {
+        final String query =
+            "SELECT " +
+                "count(nh) " +
+            "FROM " +
+                "NotificationHistory AS nh " +
+            "WHERE " +
+                "nh.endpoint.id = :endpointId " +
+            "AND " +
+                "nh.event.orgId = :orgId";
+
+        final var historyQuery = this.entityManager.createQuery(query);
+
+        historyQuery.setParameter("endpointId", endpointId);
+        historyQuery.setParameter("orgId", orgId);
+
+        return (long) historyQuery.getSingleResult();
```

Comment:
Could include the type in the query, and chain the calls to avoid the variable (not tested yet)

```suggestion
        return this.entityManager.createQuery(query, Long.class)
        .setParameter("endpointId", endpointId)
        .setParameter("orgId", orgId)
        .getSingleResult();
```

Reply:
Yeah, way better! Thanks!